src/window/layout.rs

```diff
@@ -217,6 +218,42 @@ impl Pinnacle {
         if old_mode != new_mode {
             let need_layout = old_mode.is_tiled() || new_mode.is_tiled() || new_mode.is_spilled();
 
+            match old_mode.current() {
+                crate::window::window_state::LayoutModeKind::Floating => {
+                    self.signal_state.window_unset_floating.signal(window);
+                }
+                crate::window::window_state::LayoutModeKind::Tiled => {
+                    self.signal_state.window_unset_tiled.signal(window);
+                }
+                crate::window::window_state::LayoutModeKind::Maximized => {
+                    self.signal_state.window_unset_maximized.signal(window);
+                }
+                crate::window::window_state::LayoutModeKind::Fullscreen => {
+                    self.signal_state.window_unset_fullscreen.signal(window);
+                }
+                crate::window::window_state::LayoutModeKind::Spilled => {
+                    self.signal_state.window_unset_spilled.signal(window);
+                }
+            }
+
+            match new_mode.current() {
+                crate::window::window_state::LayoutModeKind::Floating => {
+                    self.signal_state.window_set_floating.signal(window);
+                }
+                crate::window::window_state::LayoutModeKind::Tiled => {
+                    self.signal_state.window_set_tiled.signal(window);
+                }
+                crate::window::window_state::LayoutModeKind::Maximized => {
+                    self.signal_state.window_set_maximized.signal(window);
+                }
+                crate::window::window_state::LayoutModeKind::Fullscreen => {
+                    self.signal_state.window_set_fullscreen.signal(window);
+                }
+                crate::window::window_state::LayoutModeKind::Spilled => {
+                    self.signal_state.window_set_spilled.signal(window);
+                }
+            }
+
```

Comment:
This is not the only place the layout can change.

For example (there might be other instances):
- src/layout.rs:108
- src/api/window.rs:{443, 453, 459, 469}

Reply:
IMO, it's better to handle this from `State::on_event_loop_cycle_completion()` rather than sprinkle the call to `self.signal_state.window_*`.

If you need the old value, it could be held in the window `LayoutMode` until the end of the loop, then from the aforementioned function, you'd iterate over windows to clear this value and send the old/new pair.
If you don't need the old value, just a boolean would work.

This should prevent missing layout changes.

(You may want to wait for Ottatop input on that first)